pages/curriculum/[lessonSlug]/mentor/index.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useMemo, useState } from 'react'
+import {
+  AddExerciseMutation,
+  GetAppProps,
+  Module,
+  useAddExerciseMutation,
+  withGetApp
+} from '../../../../graphql'
+import { useRouter } from 'next/router'
+import { AdminLayout } from '../../../../components/admin/AdminLayout'
+import { DropdownMenu } from '../../../../components/DropdownMenu'
+import { FormCard, MD_INPUT } from '../../../../components/FormCard'
+import { formChange } from '../../../../helpers/formChange'
+import { exercisesValidation } from '../../../../helpers/formValidation'
+import ExercisePreview from '../../../../components/ExercisePreview'
+import styles from '../../../../scss/mentorPage.module.scss'
+import { get } from 'lodash'
+import { ApolloError } from '@apollo/client'
+import QueryInfo from '../../../../components/QueryInfo'
+import { errorCheckAllFields } from '../../../../helpers/admin/adminHelpers'
+import * as Sentry from '@sentry/nextjs'
+
+type HeaderProps<T> = {
+  lesson: T
+  addExerciseData?: AddExerciseMutation | null
+  loading: boolean
+  error?: ApolloError
+  setModule: (v: null | Omit<Module, 'author' | 'lesson'>) => void
+}
+const Header = <T,>({
+  lesson,
+  addExerciseData,
+  loading,
+  error,
+  setModule
+}: HeaderProps<T>) => (
+  <header className={styles.header}>
+    <div>
+      <h1>{get(lesson, 'title')}</h1>
+    </div>
+    <div className={styles.dropdownWrapper}>
+      <span>Select a module</span>
+      <DropdownMenu
+        title="Select a module"
+        items={(get(lesson, 'modules') || []).map(
+          (m: Omit<Module, 'lesson' | 'author'>) => ({
+            ...m,
+            title: m.name,
+            onClick: () => setModule({ ...m })
+          })
+        )}
+      />
+    </div>
+    <QueryInfo
+      data={addExerciseData}
+      loading={loading}
+      error={get(error, 'message') || ''}
```

Comment:
Lodash `get` takes a default value, wouldn't be better to use its default value instead of the logical `||` operator?

Reply:
Why not use `error?.message || ''` or `error?.message ?? ''`

Reply:
> Lodash get takes a default value, wouldn't be better to use its default value instead of the logical || operator?

Updated the code. Passing `get` default value worked without errors this time.

Reply:
> Why not use error?.message || '' or error?.message ?? ''

@bryanjenningz To keep things consistent, we chose to use lodash `get` instead of optional chaining.